### PR TITLE
Jp/hotfix optional gas price in rpc transaction type

### PR DESCRIPTION
**Pull request by jp-imx**
# Description

I've made `GasPrice` optional in the `transaction` struct in `jsonrpc/types.go`. The previous implementation made it impossible to call `eth_getTransactionByHash` for EIP-1559 transactions or `eth_getBlockByNumber` with the transaction details flag. on, if the block contained EIP-1559 transactions. 

# Changes include

Marking this as a HOTFIX as Immutable needs this issue solved before we can release our public testnet.

- [x] Bugfix (non-breaking change that solves an issue)
- [x] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

N/A

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

I had a local polygon-edge chain with 4 nodes. Three of them running `v1.0.0`. One of them running a version with this fix. A call to `eth_getBlockByNumber` with the transaction details flag on would fail on v1.0.0 with `nil` dereference error, but it would work on the node with the fix.

# Documentation update

N/A

# Additional comments

N/A

### CODE REVIEW

**Review by goran-ethernal — APPROVED** [review]

**Comment by jp-imx** [comment]
@goran-ethernal thanks for approving. I understand there's some flakyness in the CI, so was wondering if you think the failed PolyBFT E2E task action is unrelated to my changes. Note, I don't have access to retrying the action or merging this PR.

**Comment by goran-ethernal** [comment]
@jp-imx Nope, your changes didn't affect that test. We have fixed that test in: https://github.com/0xPolygon/polygon-edge/pull/1702, so we are good to merge this one. Just need one more approval.

**Review by dusan-maksimovic — APPROVED** [review]

**Review by igorcrevar — APPROVED** [review]

**Comment by jp-imx** [comment]
@goran-ethernal do you need me to merge that PR or maybe `develop` into my branch? Or can we merge as is? I can't retry the build though 🤔

File: jsonrpc/types.go
Comment (goran-ethernal) [inline jsonrpc/types.go]: We should add `omitempty` to `GasPrice` as well.

**Comment by goran-ethernal** [comment]
@jp-imx No need, this test has nothing to do with your changes, so we can merge this PR. I left a small comment. After that, we can merge it.

File: jsonrpc/types.go
Comment (dusan-maksimovic) [reply jsonrpc/types.go]: I was thinking about this as well. But we are not consistent with other fields either. For example: To, BlockHash, BlockNumber etc.

Comment (goran-ethernal) [reply jsonrpc/types.go]: Other fields are common to all transaction types, but `GasPrice` is not set for EIP-1559 transactions, and `GasTipCap` and `GasFeeCap` are not set for legacy transactions, so I would say only those three are optional per say.

**Review by begmaroman — APPROVED** [review]

File: jsonrpc/types.go
Comment (jp-imx) [reply jsonrpc/types.go]: I would also point out that `gasTipCap` should be `maxPriorityFeePerGas` and `gasFeeCap` should be `maxFeePerGas`. Also `chainId` is missing. Not sure if some of these are being addressed elsewhere

File: types/rlp_marshal.go
Comment (goran-ethernal) [inline types/rlp_marshal.go]: Remove this?

File: types/rlp_marshal.go
Comment (goran-ethernal) [inline types/rlp_marshal.go]: I am not sure what these changes are.

Comment (jp-imx) [reply types/rlp_marshal.go]: I'm sorry @goran-ethernal. That was a bad push 🤦 Can you have another look?

**Review by vcastellm — APPROVED** [review]

**Pull request merged**
